### ISSUE TITLE
Exec Steps Concurrent in task (task support DAG)

### ISSUE THREAD

**Issue by lugeng** [tektoncd/pipeline#3900]
### Feature request

I would exec a tekton task with multi concurrent steps, but it seems cannot support this; (I know the pipeline can handle this feature, but I think it is too heavy to move workspace in different pod); 

### Use case

![image](https://user-images.githubusercontent.com/18281825/116337909-78396280-a80d-11eb-803d-67bd6ca6ef0d.png)

In my user case, I hope that the scan step and image build step can execute concurrent.  And at last I need a job to collect the scan and build result.

**Comment by vdemeester** [comment]
@lugeng I think you might be interested by [TEP-0044: Decouple Task Composition from Scheduling](https://github.com/tektoncd/community/blob/main/teps/0044-decouple-task-composition-from-scheduling.md).

/cc @bobcatfish

**Comment by lugeng** [comment]
[PipelineRun in a Pod, running of Tasks in parallel](https://github.com/tektoncd/community/pull/318#) perfect meet our need !

**Comment by ms-lolo** [comment]
Hey just wanted to add a use case that I think is not met by being able to run pipelines on a single pod.

I have a single task that generates documentation in a repository. It's a single task because it is the one unit of reusable logic with a single output (a directory of html files).

However, my repository is made up of many python packages, and the documentation site is built by combining some top level documentation pages (a `mkdocs` site built from markdown files in a `/docs` folder) with the API documentation of each of the packages, generated by `sphinx`.

In order to generate API docs, I need to install the development environment for each of the python packages, individually, and then run my documentation command that handles running the appropriate sphinx process everywhere.

This all has to happen in the same Task because there are dependencies between some of these steps, but installing the development environment is a bit slow and should happen in parallel across all the packages in my repository. Right now, I am doing this in a single Step that runs multiple threads of my `pipenv install --dev` command. But this results in very confusing error logs when one out of the ~15 `pipenv` commands fails. If we want to get to the point where a Step is a single command, then I need to be able to have these 15 Steps run in parallel.

The output of my Task is a single html directory which I can bundle into a docker image in another task, but it doesn't make sense to have the `pipenv install` commands be their own Tasks just for the sake of parallelization. They don't represent a reusable unit of work that would be useful anywhere else.

It looks a little like this:
![image](https://user-images.githubusercontent.com/61251739/119680610-3264bf80-bdf6-11eb-9d0c-0e832fdec9f2.png)

It's worth pointing out that I really cannot run `mkdocs` or `sphinx` without all of my packages having been initialized. So there is no way to generate docs for `pkg A` above without having done `pipenv install` on all 3 packages. This is what makes turning them into multiple Tasks impossible. Docs belong to my repository (a project), not the individual components (a python package).
